### ISSUE TITLE
Implement a simple linear fan curve option

### ISSUE THREAD

**Issue by loathingKernel** [ilya-zlobintsev/LACT#415]
First of all, thank you for adding Nvidia support. I can now possibly ditch my own tool on Linux instead of making it work with Wayland.

One think I would like to have though is a simple linear fan curve based on temperature with an offset. In this mode the fan speed is continuously adjusted based on temperature using a simple formula `fan_speed = current_temperature + offset`. The only setting would be the offset. So if the temperature is `30` with an offset of `20` the fan speed would be `50`.

**Comment by ilya-zlobintsev** [comment]
I don't think that this is a good idea. In your example, using an offset of 20 with 50% fan speed at 30c you'd have way too high of a speed at lower temperatures, while having reasonable values at higher temps (100% fan speed at 80c). Lower offset would have the opposite problem.

You can replicate any behaviour you want using a fan curve. What's the reason for wanting to do this over a curve?

**Comment by loathingKernel** [comment]
>  In your example, using an offset of 20 with 50% fan speed at 30c you'd have way too high of a speed at lower temperatures, while having reasonable values at higher temps (100% fan speed at 80c). Lower offset would have the opposite problem.

In my example I used 20 but the offset would be user-adjustable so it could even go in the negatives for a reasonable range of values, for example from `-20` to `+20`. A user could also set an offset of `0`, in effect having the fan speed follow the temperature.

The main reason I would like this feature is because I like having a fine-grained increment in fan speed. As far as I understand, the current curve stays at the certain fan speed until the next threshold is reached and then it increases. A linear fan curve is not only simpler to configure but also offers a smoother fan speed transition.

**Comment by ilya-zlobintsev** [comment]
>  As far as I understand, the current curve stays at the certain fan speed until the next threshold is reached and then it increases.

This is not correct - it will pick a speed in the middle of your configured curve points to scale smoothly, it won't strictly jump between specific speeds set on your points. There are also hysteresis settings (such as the minimum change in temperature required to affect the fan speed) to avoid having rapid spinup/spindown of the fan when the temperature jumps around.
